security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -399,7 +411,26 @@ func (sc *SecretController) scrtDeleted(obj interface{}) {
 
 func (sc *SecretController) namespaceUpdated(oldObj, newObj interface{}) {
 	oldNs := oldObj.(*v1.Namespace)
-	newNs := newObj.(*v1.Namespace)
+	newNs, ok := newObj.(*v1.Namespace)
+
+	if ok {
+		rootCert := sc.ca.GetCAKeyCertBundle().GetRootCertPem()
+		certEncoded := base64.StdEncoding.EncodeToString(rootCert)
+		// Every namespaceResyncPeriod, namespaceUpdated() will be invoked
+		// for every namespace. If a namespace does not have the Citadel CA
+		// certificate or the certificate in a ConfigMap of the namespace is not
+		// up to date, Citadel updates the certificate in the namespace.
+		// For simplifying the implementation and no overhead for reading from the ConfigMap,
+		// simply updates the ConfigMap to the current Citadel CA certificate.
```

Comment:
This is misleading, InsertDataToConfigMap does read the config map?

Reply:
InsertDataToConfigMap does not read the config map. The code comment is revised as "no overhead for reading the certificate from the ConfigMap": simply writes the certificate to the ConfigMap.